Functions/xASL_stat_GetROIstatistics.m

```diff
@@ -553,7 +553,7 @@
         end
         
         % Initialize matrices with NaNs
-		FieldsAre = {'DAT_mean_PVC0' 'DAT_median_PVC0' 'DAT_mean_PVC2'}; % update this with fields filled below!
+		FieldsAre = {'DAT_mean_PVC0' 'DAT_median_PVC0'}; % update this with fields filled below!

```

Comment:
Why do you skip the initialization here? I see now that we initialize these twice, here and also below. Though note that this is just the end result (e.g., the mean ROI-value), not a mask or input data. I believe it's important to keep initializing this, to be able to differentiate missing values (NaNs in the final `*.tsv` file) with e.g., having no subject.

Also, this is not complete, as in the case you have e.g., `x.S.DAT_sum_PVC0` below (in the case of non-ASL data), this is not initialized. Note that this initialization costs no significant memory (only a matrix of subjects*ROIs).

Perhaps we can discuss a general solution for issues like this: 
1. if processing crashes, do we want a NaN output, such that the remaining pipeline can still continue?
2. if data is incomplete (too small mask or outside FoV), do we want a NaN output?

I believe the second is very important, the first is debatable. If we choose the second only, the initialization should be below, inside the loop (also easier to read the script). If we choose the first, we need to initialize everything here. Thinking of it, I would go for the second option only, removing the initialization here, but moving the initialization in the loop back to its original location

Reply:
My main motivation was that for certain cases - like whole-brain or GM/WM-masking turned of completely - it does not make sense to do PVEC. We will therefore skip the calculation. But if NaNs are prefilled, it will save a file full of NaNs. So I am skipping the result for some cases (bSkipPVC), but then I also have to skip the initialization - which I do below, but can't do it here when it is not yet know if this will be skipped.

The easiest solution is to move the NaN initialization to below only - as you say. Then I'm completely happy ;)

Regarding your question - if processing crashes then we are fine - the file is written to disc only after this function is finished, so (1) is not a problem. I agree with (2), so moving the second option below. And for cases where nothing is initialized or computed, the TSV won't be written at all - which will be fine and remove confusion.

Reply:
OK sounds good!